src/subcommand/wallet/inscribe.rs

```diff
@@ -0,0 +1,141 @@
+use super::*;
+
+use bitcoin::{
+  blockdata::{opcodes, script},
+  secp256k1::{self, rand, KeyPair, Secp256k1, XOnlyPublicKey},
+  util::sighash::{Prevouts, SighashCache},
+  util::taproot::TaprootBuilder,
+  util::taproot::{LeafVersion, TapLeafHash},
+  PackedLockTime, SchnorrSighashType, Witness,
+};
+
+#[derive(Debug, Parser)]
+pub(crate) struct Inscribe {
+  ordinal: Ordinal,
+  content: String,
+}
+
+impl Inscribe {
+  pub(crate) fn run(self, options: Options) -> Result {
```

Comment:
Do as much as possible before sending any transaction. All setup should be done, signing, construction, etc. and final step is to broadcast both transactions.